api/v1alpha1/ndb_api_helpers.go

```diff
@@ -155,43 +157,147 @@ func GetNoneTimeMachineSLA(ctx context.Context, ndbclient *ndbclient.NDBClient)
 	return sla, fmt.Errorf("NONE TimeMachine not found")
 }
 
-// Fetches all the profiles and returns a map of OOB profiles
-// Returns an error if one or more OOB profiles is not found
-func GetOOBProfiles(ctx context.Context, ndbclient *ndbclient.NDBClient, dbType string) (profileMap map[string]ProfileResponse, err error) {
-	// Map of profile type to profiles
-	profileMap = make(map[string]ProfileResponse)
+// +kubebuilder:object:generate:=false
+type ProfileResolver interface {
+	Resolve(ctx context.Context, allProfiles []ProfileResponse, pType string, filter func(p ProfileResponse) bool) (profile ProfileResponse, err error)
+}
```

Comment:
Since its type just a string (as opposed to a custom defined struct), I think "type" would not concretly convey what it refers to (pType vs dbType vs any other type that might be applicable). pType is supposed to imply its a "profileType". Since I do not want to bloat the signature size, I think pType should be reasonable, the name "type" sounds too generic.

Reply:
Alright, then how about being completely explicit: `profileType`? Just thinking about readability here..